pkg/querier/queryrange/codec.go

```diff
@@ -2125,7 +2086,81 @@ func mergeLokiResponse(responses ...queryrangebase.Response) *LokiResponse {
 	}
 }
 
-// In some other world LabelRequest could implement queryrangebase.Request.
+func parseRangeQueryWithStoreChunksExtension(r *http.Request) (*LokiRequest, error) {
```

Comment:
I don't see this method being called conditionally of having chunks. I think this as well as the instant should be renamed to `parseRangeQuery` and `parseInstantQuery`